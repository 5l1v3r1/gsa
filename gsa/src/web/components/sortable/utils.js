/* Greenbone Security Assistant
 *
 * Authors:
 * Bj√∂rn Ricks <bjoern.ricks@greenbone.net>
 *
 * Copyright:
 * Copyright (C) 2018 Greenbone Networks GmbH
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import uuid from 'uuid/v4';

import {createRow} from 'gmp/commands/dashboards';

export const createItem = (props, uuidFunc = uuid) => {
  const id = uuidFunc();

  return {
    id,
    ...props,
  };
};

export const removeItem = (rows, itemId) => rows.map(row => ({
  ...row,
  items: row.items.filter(item => item.id !== itemId),
})).filter(row => row.items.length > 0);

export const updateRow = (row, data) => ({
  ...row,
  ...data,
});

export const convertDefaultContent = (defaultContent = [], uuidFunc = uuid) => {
  return {
    rows: defaultContent.map(row => createRow(
      row.map(item => createItem({name: item}, uuidFunc)), undefined, uuidFunc
    )),
  };
};

// vim: set ts=2 sw=2 tw=80:
