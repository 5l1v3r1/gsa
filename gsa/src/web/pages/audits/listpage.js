/* Copyright (C) 2016-2019 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: GPL-2.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import React from 'react';

import _ from 'gmp/locale';

import {TASKS_FILTER_FILTER} from 'gmp/models/filter';

import PropTypes from 'web/utils/proptypes';
import withCapabilities from 'web/utils/withCapabilities';

import {
  loadEntities,
  selector as entitiesSelector,
} from 'web/store/entities/tasks';

import EntitiesPage from 'web/entities/page';
import withEntitiesContainer from 'web/entities/withEntitiesContainer';

// import DashboardControls from 'web/components/dashboard/controls';

import ManualIcon from 'web/components/icon/manualicon';

import IconDivider from 'web/components/layout/icondivider';

import {DEFAULT_RELOAD_INTERVAL_ACTIVE} from 'web/utils/constants';

// import NewIconMenu from 'web/pages/tasks/icons/newiconmenu';

import AuditComponent from './component';
// import TaskDashboard, {TASK_DASHBOARD_ID} from './dashboard';
import TaskFilterDialog from './filterdialog';
import Table from './table';
import TaskIcon from 'web/components/icon/taskicon';

const ToolBarIcons = withCapabilities(({capabilities}) => (
  <IconDivider>
    <ManualIcon
      page="vulnerabilitymanagement"
      anchor="creating-a-task"
      title={_('Help: Tasks')}
    />
  </IconDivider>
));

/* ToolBarIcons.propTypes = {
  onAdvancedTaskWizardClick: PropTypes.func.isRequired,
  onContainerTaskCreateClick: PropTypes.func.isRequired,
  onModifyTaskWizardClick: PropTypes.func.isRequired,
  onTaskCreateClick: PropTypes.func.isRequired,
  onTaskWizardClick: PropTypes.func.isRequired,
}; */

const Page = ({
  filter,
  onFilterChanged,
  onInteraction,
  onChanged,
  onDownloaded,
  onError,
  ...props
}) => (
  <AuditComponent
    // onAdvancedTaskWizardSaved={onChanged}
    onCloned={onChanged}
    onCloneError={onError}
    // onContainerSaved={onChanged}
    // onCreated={onChanged}
    // onContainerCreated={onChanged}
    onDeleted={onChanged}
    onDeleteError={onError}
    onDownloaded={onDownloaded}
    onDownloadError={onError}
    onInteraction={onInteraction}
    // onModifyTaskWizardSaved={onChanged}
    // on Report Dowloaded???
    onReportImported={onChanged}
    onResumed={onChanged}
    onResumeError={onError}
    onSaved={onChanged}
    onStarted={onChanged}
    onStartError={onError}
    onStopped={onChanged}
    onStopError={onError}
    // onTaskWizardSaved={onChanged}
  >
    {({
      clone,
      // create,
      // createcontainer,
      delete: delete_func,
      download,
      edit,
      start,
      stop,
      resume,
      reportDownload,
      reportimport,
      gcrFormatDefined,
      // advancedtaskwizard,
      // modifytaskwizard,
      // taskwizard,
    }) => (
      <EntitiesPage
        {...props}
        /* dashboard={() => (
          <TaskDashboard
            filter={filter}
            onFilterChanged={onFilterChanged}
            onInteraction={onInteraction}
          />
        )}
        dashboardControls={() => (
          <DashboardControls
            dashboardId={TASK_DASHBOARD_ID}
            onInteraction={onInteraction}
          />
        )} */
        filter={filter}
        filterEditDialog={TaskFilterDialog}
        filtersFilter={TASKS_FILTER_FILTER}
        gcrFormatDefined={gcrFormatDefined}
        sectionIcon={<TaskIcon size="large" />}
        table={Table}
        title={_('Audits')}
        toolBarIcons={ToolBarIcons}
        // onAdvancedTaskWizardClick={advancedtaskwizard}
        // onContainerTaskCreateClick={createcontainer}
        onError={onError}
        onFilterChanged={onFilterChanged}
        onInteraction={onInteraction}
        // onModifyTaskWizardClick={modifytaskwizard}
        onReportDownloadClick={reportDownload}
        onReportImportClick={reportimport}
        onTaskCloneClick={clone}
        // onTaskCreateClick={create}
        onTaskDeleteClick={delete_func}
        onTaskDownloadClick={download}
        onTaskEditClick={edit}
        onTaskResumeClick={resume}
        onTaskStartClick={start}
        onTaskStopClick={stop}
        // onTaskWizardClick={taskwizard}
      />
    )}
  </AuditComponent>
);

Page.propTypes = {
  filter: PropTypes.filter,
  onChanged: PropTypes.func.isRequired,
  onDownloaded: PropTypes.func.isRequired,
  onError: PropTypes.func.isRequired,
  onFilterChanged: PropTypes.func.isRequired,
  onInteraction: PropTypes.func.isRequired,
};

const taskReloadInterval = ({entities = [], defaultReloadInterval}) =>
  entities.some(task => task.isActive())
    ? DEFAULT_RELOAD_INTERVAL_ACTIVE
    : defaultReloadInterval;

export default withEntitiesContainer('task', {
  entitiesSelector,
  loadEntities,
  reloadInterval: taskReloadInterval,
})(Page);

// vim: set ts=2 sw=2 tw=80:
