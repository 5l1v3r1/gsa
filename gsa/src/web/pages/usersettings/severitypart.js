/* Greenbone Security Assistant
 *
 * Authors:
 * Steffen Waterkamp <steffen.waterkamp@greenbone.net>
 *
 * Copyright:
 * Copyright (C) 2018 Greenbone Networks GmbH
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */

import React from 'react';

import _ from 'gmp/locale';
import {parseYesNo, YES_VALUE, NO_VALUE} from 'gmp/parser';

import Checkbox from 'web/components/form/checkbox';
import FormGroup from 'web/components/form/formgroup';
import Select from 'web/components/form/select';
import Spinner from 'web/components/form/spinner';

import PropTypes from 'web/utils/proptypes';
import {renderSelectItems} from 'web/utils/render';

import {SEVERITY_CLASSES} from './usersettingspage';

const SeverityPart = ({
  defaultSeverity,
  dynamicSeverity,
  severityClass,
  onChange,
}) => {
  return (
    <React.Fragment>
      <FormGroup title={_('Severity Class')} titleSize="3">
        <Select
          name="severityClass"
          value={severityClass}
          items={renderSelectItems(SEVERITY_CLASSES)}
          onChange={onChange}
        />
      </FormGroup>
      <FormGroup title={_('Dynamic Severity')} titleSize="3">
        <Checkbox
          name="dynamicSeverity"
          checked={parseYesNo(dynamicSeverity) === YES_VALUE}
          checkedValue={YES_VALUE}
          unCheckedValue={NO_VALUE}
          onChange={onChange}
        />
      </FormGroup>
      <FormGroup title={_('Default Severity')} titleSize="3">
        <Spinner
          name="defaultSeverity"
          value={defaultSeverity}
          min="0"
          max="10"
          step="0.1"
          type="float"
          onChange={onChange}/>
      </FormGroup>
    </React.Fragment>
  );
};

SeverityPart.propTypes = {
  defaultSeverity: PropTypes.number,
  dynamicSeverity: PropTypes.string,
  severityClass: PropTypes.string,
  onChange: PropTypes.func,
};

export default SeverityPart;

// vim: set ts=2 sw=2 tw=80:
