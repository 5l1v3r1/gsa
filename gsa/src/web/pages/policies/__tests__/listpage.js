/* Copyright (C) 2019 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: GPL-2.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */
import React from 'react';

import Capabilities from 'gmp/capabilities/capabilities';
import CollectionCounts from 'gmp/collection/collectioncounts';

import Filter from 'gmp/models/filter';
import Policy from 'gmp/models/policy';
import {OPENVAS_SCAN_CONFIG_TYPE} from 'gmp/models/scanconfig';

import {setUsername} from 'web/store/usersettings/actions';

import {entitiesActions} from 'web/store/entities/audits';
import {rendererWith, fireEvent} from 'web/utils/testing';

import PoliciesPage, {ToolBarIcons} from '../listpage';

const policy = new Policy({
  _id: '12345',
  name: 'foo',
  comment: 'bar',
  creation_time: '2019-07-16T06:31:29Z',
  modification_time: '2019-07-16T06:44:55Z',
  owner: {name: 'admin'},
  writable: '1',
  in_use: '0',
  usage_type: 'policy',
  permissions: {permission: [{name: 'everything'}]},
  scanner: {name: 'scanner', type: '42'},
  type: OPENVAS_SCAN_CONFIG_TYPE,
  tasks: {
    task: [{id: '1234', name: 'audit1'}, {id: '5678', name: 'audit2'}],
  },
});

const caps = new Capabilities(['everything']);
const wrongCaps = new Capabilities(['get_config']);

const reloadInterval = 1;
const manualUrl = 'test/';

const currentSettings = jest.fn().mockReturnValue(
  Promise.resolve({
    foo: 'bar',
  }),
);

const getFilters = jest.fn().mockReturnValue(
  Promise.resolve({
    data: [],
    meta: {
      filter: Filter.fromString(),
      counts: new CollectionCounts(),
    },
  }),
);

const getPolicies = jest.fn().mockReturnValue(
  Promise.resolve({
    data: [policy],
    meta: {
      filter: Filter.fromString(),
      counts: new CollectionCounts(),
    },
  }),
);

describe('PoliciesPage tests', () => {
  test('should render full PoliciesPage', () => {
    const gmp = {
      policies: {
        get: getPolicies,
      },
      filters: {
        get: getFilters,
      },
      reloadInterval,
      settings: {manualUrl},
      user: {currentSettings},
    };

    const {render, store} = rendererWith({
      gmp,
      capabilities: true,
      store: true,
      router: true,
    });

    store.dispatch(setUsername('admin'));

    const counts = new CollectionCounts({
      first: 1,
      all: 1,
      filtered: 1,
      length: 1,
      rows: 10,
    });
    const filter = Filter.fromString('first=1 rows=10');
    const loadedFilter = Filter.fromString('first=1 rows=10');
    store.dispatch(
      entitiesActions.success([policy], filter, loadedFilter, counts),
    );

    const {baseElement} = render(<PoliciesPage />);

    expect(baseElement).toMatchSnapshot();
  });
});

describe('PoliciesPage ToolBarIcons test', () => {
  test('should render', () => {
    const handlePolicyCreateClick = jest.fn();
    const handlePolicyImportClick = jest.fn();

    const {render} = rendererWith({
      capabilities: caps,
      router: true,
    });

    const {element} = render(
      <ToolBarIcons
        entity={policy}
        onPolicyCreateClick={handlePolicyCreateClick}
        onPolicyImportClick={handlePolicyImportClick}
      />,
    );
    expect(element).toMatchSnapshot();
  });

  test('should call click handlers', () => {
    const handlePolicyCreateClick = jest.fn();
    const handlePolicyImportClick = jest.fn();

    const {render} = rendererWith({
      capabilities: caps,
      router: true,
    });

    const {getAllByTestId} = render(
      <ToolBarIcons
        entity={policy}
        onPolicyCreateClick={handlePolicyCreateClick}
        onPolicyImportClick={handlePolicyImportClick}
      />,
    );

    const icons = getAllByTestId('svg-icon');

    fireEvent.click(icons[0]);
    expect(handlePolicyCreateClick).toHaveBeenCalled();
    expect(icons[0]).toHaveAttribute('title', 'New Policy');

    fireEvent.click(icons[1]);
    expect(handlePolicyImportClick).toHaveBeenCalled();
    expect(icons[1]).toHaveAttribute('title', 'Import Policy');
  });

  test('should not call click handlers without permission', () => {
    const handlePolicyCreateClick = jest.fn();
    const handlePolicyImportClick = jest.fn();

    const {render} = rendererWith({
      capabilities: wrongCaps,
      router: true,
    });

    const {queryAllByTestId} = render(
      <ToolBarIcons
        onPolicyCreateClick={handlePolicyCreateClick}
        onPolicyImportClick={handlePolicyImportClick}
      />,
    );

    const icons = queryAllByTestId('svg-icon');
    expect(icons.length).toBe(0);
  });
});
