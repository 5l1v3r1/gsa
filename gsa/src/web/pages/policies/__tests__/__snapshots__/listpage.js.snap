// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoliciesPage ToolBarIcons test should render 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -5px;
}

.c1 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 > * {
  margin-left: 5px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c3 {
  cursor: pointer;
}

.c2 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c2 * {
  height: inherit;
  width: inherit;
}

@media print {
  .c3 {
    display: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    margin="5px"
  >
    <span
      class="c2 c3"
      data-testid="svg-icon"
      title="New Policy"
    >
      <svg>
        new.svg
      </svg>
    </span>
    <span
      class="c2 c3"
      data-testid="svg-icon"
      title="Import Policy"
    >
      <svg>
        upload.svg
      </svg>
    </span>
  </div>
</div>
`;

exports[`PoliciesPage tests should render full PoliciesPage 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stetch;
  -webkit-box-align: stetch;
  -ms-flex-align: stetch;
  align-items: stetch;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -5px;
}

.c3 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c3 > * {
  margin-left: 5px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -5px;
}

.c10 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c10 > * {
  margin-left: 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c5 {
  cursor: pointer;
}

.c4 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c4 * {
  height: inherit;
  width: inherit;
}

.c25 {
  height: 50px;
  width: 50px;
  line-height: 50px;
}

.c25 * {
  height: inherit;
  width: inherit;
}

.c28 {
  width: 80px;
  height: 80px;
  margin: 40px auto;
  background-image: url(greenbone.svg);
  background-size: 90%;
  background-position: center;
  background-repeat: no-repeat;
  -webkit-animation: fSFzYT 2s infinite ease-in-out;
  animation: fSFzYT 2s infinite ease-in-out;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin: 10px 0px;
  padding-bottom: 1px;
  border-bottom: 2px solid #c8d3d9;
  position: relative;
}

.c23 {
  margin: 0 0 1px 0;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-right: 5px;
}

.c26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  word-break: break-all;
  min-width: 100px;
}

.c20 {
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  margin: 1px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c19 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c19 * {
  height: inherit;
  width: inherit;
}

.c16 {
  border: 1px solid #aaaaaa;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 1px 5px;
  background-color: #fff;
  color: #000;
  font-weight: normal;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 150px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  word-break: keep-all;
  white-space: nowrap;
  overflow: hidden;
  cursor: pointer;
  cursor: default;
}

.c14 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  display: block;
  height: 22px;
  color: #393637;
  background-color: #fff;
  background-image: none;
  border: 1px solid #aaaaaa;
  border-radius: 2px;
  padding: 1px 8px;
}

.c14:-webkit-autofill {
  box-shadow: 0 0 0 1000px white inset;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 {
  margin-right: 5px;
}

.c9 {
  margin-right: 5px;
}

@media print {
  .c5 {
    display: none;
  }
}

<body>
  <div
    id="portals"
  />
  <div>
    <div
      class="c0"
    >
      <div
        class="toolbar c1"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
            margin="5px"
          >
            <span
              class="c4 c5"
              data-testid="svg-icon"
              title="New Policy"
            >
              <svg>
                new.svg
              </svg>
            </span>
            <span
              class="c4 c5"
              data-testid="svg-icon"
              title="Import Policy"
            >
              <svg>
                upload.svg
              </svg>
            </span>
          </div>
        </div>
        <div
          class="c6"
        >
          <div
            class="powerfilter c7"
          >
            <div
              class="c8"
            >
              <div
                class="c2"
              >
                <div
                  class="c9 c10"
                  margin="5px"
                >
                  <div
                    class="c11"
                  >
                    <label
                      class="c12"
                    >
                      <b>
                        Filter
                      </b>
                    </label>
                    <input
                      class="c13 c14"
                      maxlength="1000"
                      name="userFilterString"
                      size="53"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="c2"
                  >
                    <div
                      class="c10"
                      margin="5px"
                    >
                      <span
                        class="c4 c5"
                        data-testid="svg-icon"
                        title="Update Filter"
                      >
                        <svg>
                          refresh.svg
                        </svg>
                      </span>
                      <span
                        class="c4 c5"
                        data-testid="svg-icon"
                        title="Remove Filter"
                      >
                        <svg>
                          delete.svg
                        </svg>
                      </span>
                      <span
                        class="c4 c5"
                        data-testid="svg-icon"
                        title="Reset to Default Filter"
                      >
                        <svg>
                          reset.svg
                        </svg>
                      </span>
                      <a
                        href="test/en/web-interface.html#filtering-the-page-content"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        <span
                          class="c4 "
                          data-testid="svg-icon"
                          title="Help: Powerfilter"
                        >
                          <svg>
                            help.svg
                          </svg>
                        </span>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-0-label"
                class="c15"
                role="combobox"
                width="150px"
              >
                <div
                  class="c16"
                  title="Loaded filter"
                >
                  <div
                    class="c17"
                    data-testid="select-selected-value"
                    title="Loaded filter"
                  >
                    --
                  </div>
                  <div
                    class="c18"
                  >
                    <span
                      aria-haspopup="true"
                      aria-label="open menu"
                      class="c19 c20"
                      data-testid="select-open-button"
                      data-toggle="true"
                      role="button"
                      type="button"
                    >
                      ▼
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <section
        class="entities-section"
      >
        <div
          class="section-header c21"
        >
          <h2
            class="c22 c23"
          >
            <div
              class="c24"
            >
              <span
                class="c25 "
                data-testid="svg-icon"
              >
                <svg>
                  policy.svg
                </svg>
              </span>
            </div>
            <div
              class="c26"
            >
              Policies
            </div>
          </h2>
          <div
            class="c8"
          />
        </div>
        <div
          class="c0"
        >
          <div
            class="c27"
          >
            <div
              class="c28"
            />
          </div>
        </div>
      </section>
    </div>
    <a
      aria-hidden="true"
      style="display: none;"
    >
      Download
    </a>
  </div>
</body>
`;
